app/models/user.model.js

```diff
@@ -0,0 +1,28 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcryptjs');
+
+const userSchema = new mongoose.Schema({
+  firstName: String,
+  lastName: String,
+  phoneNumber: String,
+  email: String,
+  password: String,
+});
+
+// Hash the password before saving to the database
+userSchema.pre('save', async function (next) {
+  try {
+    if (!this.isModified('password')) {
+      return next();
+    }
+    const hashedPassword = await bcrypt.hash(this.password, 10);
+    this.password = hashedPassword;
+    next();
+  } catch (err) {
+    next(err);
+  }
+});
+
+const User = mongoose.model('User', userSchema);
+
+module.exports = User;
```

Comment:
rename this from user.model to user.js only as file naming should not contain `.`